File: src/IDs/SessionID.java
package IDs;

//Samet CAN 150120528
public class SessionID implements UniqueID {

	private int id;

	public SessionID(int id) {
		super();
		this.id = id;
	}

	// Another set method for ID
	public void setID(String string) {
		try {
			Integer.parseInt(string);
		} catch (NumberFormatException e) {

		} finally {
			this.id = Integer.parseInt(string);
		}

	}

	// Set and get methods for ID
	public void setID(int Id) {
		this.id = Id;
	}

	public String getID() {
		return id + "";
	}

}

File: Iteration 3/pythonCode/SessionID.py
from UniqueID import UniqueID

class SessionID(UniqueID):
    def __init__(self, id: int):
        self.__id = id

    # Creating properties for variables

    def getID(self) -> str:
        return str(self.__id)

    def setID(self, *args):
        t = args
        if (isinstance(t, str)):
            try:
                int(t)
            except ValueError:
                pass
            finally:
                self.__id = int(t)

        if t == int:
                self.__id = t
